File: C언어/SWEA/3349/SWEA_3349.c
#include <stdio.h>

int main(void) {
	int T;

	scanf("%d", &T);

	for (int tc = 1; tc <= T; tc++) {
		int W, H, N, x, y, nx, ny, dx, dy, ans = 0;
		scanf("%d %d %d", &W, &H, &N);
		scanf("%d %d", &x, &y);

		for (int i = 1; i < N; i++) {
			scanf("%d %d", &nx, &ny);
			dx = nx - x;
			dy = ny - y;

			if (dx * dy >= 0) {
				if (abs(dx) >= abs(dy)) {
					ans += abs(dx);
				}
				else {
					ans += abs(dy);
				}
			}
			else {
				ans += abs(dx) + abs(dy);
			}

			x = nx;
			y = ny;

		}
		printf("#%d %d\n", tc, ans);
	}
	return 0;
}

File: Python/SWEA/SWEA_3349.py
for tc in range(1, int(input()) + 1):
    W, H, N = map(int,input().split())
    x, y = map(int,input().split())

    ans = 0
    for _ in range(N-1):
        nx, ny = map(int,input().split())
        dx = nx - x
        dy = ny - y
        if dx * dy >= 0:
            ans += max(abs(dx), abs(dy))
        else:
            ans += abs(dx) + abs(dy)
        x, y = nx, ny

    print("#%d"%(tc), ans)